builtin/logical/pki/storage.go

```diff
@@ -1235,7 +1236,32 @@ func (sc *storageContext) writeAutoTidyConfig(config *tidyConfig) error {
 		return err
 	}
 
-	return sc.Storage.Put(sc.Context, entry)
+	err = sc.Storage.Put(sc.Context, entry)
+	if err != nil {
+		return err
+	}
+
+	sc.Backend.publishCertCountMetrics.Store(config.PublishMetrics)
+
+	// To Potentially Disable Certificate Counting
+	if config.MaintainCount == false {
+		oldValue := sc.Backend.certCountEnabled.Swap(config.MaintainCount)
+		if oldValue != false {
+			if config.MaintainCount == false {
```

Comment:
I don't understand the need to recheck `config.MaintainCount` here?

Also I think this would be easier to read than parsing in my head the negation of false?

```
certCountWasEnabled := sc.Backend.certCountEnabled.Swap(config.MaintainCount)
if certCountWasEnabled {
```